recipe-server/client/control/components/Fields.js

```diff
@@ -39,10 +40,65 @@ export function buildControlField({
 buildControlField.propTypes = {
   input: pt.object.isRequired,
   meta: pt.shape({
-    error: pt.string,
+    error: pt.oneOfType([pt.string, pt.array]),
   }).isRequired,
   label: pt.string.isRequired,
   className: pt.string,
   InputComponent: pt.oneOfType([pt.func, pt.string]),
+  hideErrors: pt.bool,
   children: pt.node,
 };
+
+export function IntegerControlField(props) {
+  return (
+    <ControlField
+      component="input"
+      type="number"
+      step="1"
+      parse={value => {
+        try {
+          return Number.parseInt(value, 10);
+        } catch (err) {
+          return 0;
+        }
+      }}
+      {...props}
+    />
+  );
+}
+
+export function BooleanRadioControlField(props) {
+  return (
+    <ControlField
+      component="input"
+      type="radio"
+      className="radio-field"
+      parse={value => value === 'true'}
+      format={value => {
+        if (value) {
+          return 'true';
```

Comment:
The booleans-as-strings business seems funky to me, but I'm assuming it's a redux-form thing?